### PR TITLE
BACKLOG-10137 Restoring Data Service references when restoring transformations from trash

### PR DESCRIPTION
**Pull request by flbrino**
BACKLOG-10137 Restoring Data Service references when restoring transformations from trash

### CODE REVIEW

**Comment by flbrino** [comment]
@mkambol for validation.

File: plugins/pdi-pur-plugin/src/org/pentaho/di/ui/repository/pur/repositoryexplorer/controller/TrashBrowseController.java [hunk 1]
```diff
@@ -384,6 +388,27 @@ public void undelete() {
...
+          // if transformation or directory with transformations call extension to restore data services references.
+          if ( RepositoryObjectType.TRANSFORMATION.name().equals( uiObj.getType() ) ) {
+            TransMeta transMeta = repository.loadTransformation( uiObj.getId(), null );
+            ExtensionPointHandler
+                .callExtensionPoint( LogChannel.GENERAL, KettleExtensionPoint.TransAfterOpen.id, transMeta );
+            transMeta.clearChanged();
+          } else if ( !RepositoryObjectType.JOB.name().equals( uiObj.getType() ) ) {
+            // if not a transformation and not a job then is a Directory
+            RepositoryDirectoryInterface
+                actualDir =
```
Comment (mkambol) [inline plugins/pdi-pur-plugin/src/org/pentaho/di/ui/repository/pur/repositoryexplorer/controller/TrashBrowseController.java]: If uiObj is not a directory, or is not found, .findDirectory() will return null.
We should check for null and handle it.

Comment (flbrino) [inline plugins/pdi-pur-plugin/src/org/pentaho/di/ui/repository/pur/repositoryexplorer/controller/TrashBrowseController.java]: Done. New commit

**Comment by flbrino** [comment]
@mkambol Check and handle null in case of directory not found. For validation

**Pull request merged**